tools/percolator/metafiles2pin.xml

```diff
@@ -0,0 +1,110 @@
+<tool id="metafiles2pin" name="Merge search data" version="1.0">
+    <description>before creating percolator input</description>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+    <command interpreter="python">metafiles2pin.py
+        --batchsize $percopoolsize
+        #if len($percopoolids) > 0
+        --percolator-pool-ids
+        #for $ppool in $percopoolids
+        "${ppool.ppool_identifier}"
+        #end for
+        #end if
+        --spectrafiles
+        #for $key in $searchresult.keys()
```

Comment:
If I remove .keys() it doesn’t work. Errors when generating job:

galaxy.jobs.runners: ERROR: (12) Failure preparing job
Traceback (most recent call last):
  File "/mnt/galaxy/galaxy-dist/lib/galaxy/jobs/runners/__init__.py", line 166, in prepare_job
    job_wrapper.prepare()
  File "/mnt/galaxy/galaxy-dist/lib/galaxy/jobs/__init__.py", line 877, in prepare
    self.command_line, self.extra_filenames, self.environment_variables = tool_evaluator.build()
  File "/mnt/galaxy/galaxy-dist/lib/galaxy/tools/evaluation.py", line 429, in build
    raise e
KeyError: <galaxy.tools.wrappers.DatasetFilenameWrapper object at 0x7efbf3e23210>

Difference between cheetah and python maybe?

cheers,
— 
Jorrit Boekel
Proteomics systems developer
BILS / Lehtiö lab
Scilifelab Stockholm, Sweden



> On 15 Apr 2016, at 15:10, Björn Grüning <notifications@github.com> wrote:
> 
> In tools/percolator/metafiles2pin.xml:
> 
> > @@ -0,0 +1,110 @@
> > +<tool id="metafiles2pin" name="Merge search data" version="1.0">
> > +    <description>before creating percolator input</description>
> > +    <stdio>
> > +        <exit_code range="1:" />
> > +    </stdio>
> > +    <command interpreter="python">metafiles2pin.py
> > +        --batchsize $percopoolsize
> > +        #if len($percopoolids) > 0
> > +        --percolator-pool-ids
> > +        #for $ppool in $percopoolids
> > +        "${ppool.ppool_identifier}"
> > +        #end for
> > +        #end if
> > +        --spectrafiles
> > +        #for $key in $searchresult.keys()
> 
> Sorry, #for $file ...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>